apps/web/pages/auth/setup/index.tsx

```diff
@@ -1,39 +1,86 @@
 import { UserPermissionRole } from "@prisma/client";
+import { DeploymentLicenseType } from "@prisma/client";
 import { GetServerSidePropsContext } from "next";
-import { useState } from "react";
+import { Dispatch, SetStateAction, useState } from "react";
 
 import AdminAppsList from "@calcom/features/apps/AdminAppsList";
 import { getSession } from "@calcom/lib/auth";
 import { useLocale } from "@calcom/lib/hooks/useLocale";
 import prisma from "@calcom/prisma";
 import { inferSSRProps } from "@calcom/types/inferSSRProps";
-import { WizardForm } from "@calcom/ui";
+import { Meta, WizardForm } from "@calcom/ui";
 
-import SetupFormStep1 from "./steps/SetupFormStep1";
+import AdminUser from "./steps/AdminUser";
+import ChooseLicense from "./steps/ChooseLicense";
+import EnterpriseLicense from "./steps/EnterpriseLicense";
 import StepDone from "./steps/StepDone";
 
 export default function Setup(props: inferSSRProps<typeof getServerSideProps>) {
   const { t } = useLocale();
-  const [isLoadingStep1, setIsLoadingStep1] = useState(false);
-  const shouldDisable = props.userCount !== 0;
+  const [isFreeLicense, setIsFreeLicense] = useState(
+    props.deployment?.licenseType !== DeploymentLicenseType.EE
+  );
+  const [isEnabledEE, setIsEnabledEE] = useState(!!props.deployment?.licenseKey);
 
   const steps = [
     {
       title: t("administrator_user"),
       description: t("lets_create_first_administrator_user"),
-      content: shouldDisable ? <StepDone /> : <SetupFormStep1 setIsLoading={setIsLoadingStep1} />,
-      isLoading: isLoadingStep1,
+      contentEl: (setIsLoading: Dispatch<SetStateAction<boolean>>) =>
+        props.userCount !== 0 ? (
+          <StepDone currentStep={1} nextStepPath="/auth/setup?step=2" setIsLoading={setIsLoading} />
+        ) : (
+          <AdminUser setIsLoading={setIsLoading} />
+        ),
+    },
+    {
+      title: t("choose_a_license"),
+      description: t("choose_license_description"),
+      contentEl: (setIsLoading: Dispatch<SetStateAction<boolean>>) =>
+        props.deployment?.licenseType ? (
+          <StepDone currentStep={2} nextStepPath="/auth/setup?step=3" setIsLoading={setIsLoading} />
+        ) : (
```

Comment:
What' s the point of having a "prev step" if we cannot do anything about it?

![image](https://user-images.githubusercontent.com/3504472/214646485-e8b9ccdf-9cfa-45bb-85c7-277fa1a72641.png)

IMO we should allow to go back and change the license. Makes sense to not do that with the first admin user creation tho.

Reply:
Done!